GVFS/GVFS.Common/LibGit2RepoPool.cs

```diff
@@ -9,24 +9,38 @@ namespace GVFS.Common
     public class LibGit2RepoPool
     {
         private const int TryAddTimeoutMilliseconds = 10;
-        private const int TryTakeTimeoutMilliseconds = Timeout.Infinite;
+        private const int TryTakeWaitForever = Timeout.Infinite;
+        private const int TryTakeWaitShort = 10;
 
         private readonly BlockingCollection<LibGit2Repo> pool;
+        private readonly Func<LibGit2Repo> repoFactory;

```

Comment:
Maybe name the function with what it exactly - `OpenRepository`, `CreateLibGit2Repo`?

Reply:
It seems to be purposefully generic, but I'll rename it to what the constructor uses as a parameter.